flink-ai-flow/ai_flow_plugins/scheduler_plugins/airflow/ai_flow_operator.py

```diff
@@ -0,0 +1,105 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Any, Text
+import os
+from airflow.models.baseoperator import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from ai_flow.common.module_load import import_string
+from ai_flow.util.time_utils import datetime_to_int64
+from ai_flow.plugin_interface.blob_manager_interface import BlobManagerFactory
+from ai_flow.plugin_interface.job_plugin_interface import JobController, JobHandle, JobRuntimeEnv
+from ai_flow.plugin_interface.scheduler_interface import JobExecutionInfo, WorkflowExecutionInfo, WorkflowInfo
+from ai_flow.context.project_context import build_project_context
+from ai_flow.workflow.job import Job
+from ai_flow.workflow.workflow import Workflow, WorkflowPropertyKeys
+from ai_flow_plugins.job_plugins.job_utils import prepare_job_runtime_env
+
+
+class AIFlowOperator(BaseOperator):
+    @apply_defaults
+    def __init__(
+            self,
+            job: Job,
+            workflow: Workflow,
+            **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.job: Job = job
+        self.workflow: Workflow = workflow
+        plugins = self.workflow.properties.get(WorkflowPropertyKeys.JOB_PLUGINS)
+        module, name = plugins.get(self.job.job_config.job_type)
+        class_object = import_string('{}.{}'.format(module, name))
+        self.job_controller: JobController = class_object()
+        self.job_handler: JobHandle = None
+        self.job_runtime_env: JobRuntimeEnv = None
+
+    def context_to_job_info(self, project_name: Text, context: Any) -> JobExecutionInfo:
```

Comment:
The `context_to_job_info` method could not bind with the context from Airflow. The AIFlowOperator should be not about the scheduler.